src/ScriptCoverageManager.cc

```diff
@@ -127,31 +124,48 @@ bool ScriptCoverageManager::WriteStats()
 		return false;
 		}
 
-	for ( auto s : stmts )
+	for ( auto& s : stmts )
 		{
-		ODesc location_info;
-		s->GetLocationInfo()->Describe(&location_info);
 		ODesc desc_info;
 		s->Describe(&desc_info);
 		string desc(desc_info.Description());
-		canonicalize_desc cd{delim};
-		for_each(desc.begin(), desc.end(), cd);
-		pair<string, string> location_desc(location_info.Description(), desc);
-		if ( usage_map.find(location_desc) != usage_map.end() )
-			usage_map[location_desc] += s->GetAccessCount();
-		else
-			usage_map[location_desc] = s->GetAccessCount();
+		TrackUsage(s, desc_info.Description(), s->GetAccessCount());
 		}
 
-	map<pair<string, string>, uint64_t>::const_iterator it;
-	for ( auto& um : usage_map )
+	for ( auto& fi : func_instances )
 		{
-		fprintf(f, "%" PRIu64 "%c%s%c%s\n", um.second, delim, um.first.first.c_str(), delim,
-		        um.first.second.c_str());
+		auto& func = fi.first;
+		auto& body = fi.second;
+
+		auto ft = func->GetType<FuncType>();
+		auto desc = ft->FlavorString() + " " + func->Name() + " TOTAL";
+
+		TrackUsage(body, desc, body->GetAccessCount());
 		}
 
+	for ( auto& um : usage_map )
```

Comment:
This could do the same as above with a structured binding to reduce the `um.first` and `um.second` usage, plus be a bit more descriptive. Something like:

```suggestion
	for ( auto& [location_info, count] : usage_map )
```